### ISSUE TITLE
Remove unnecessary code in unit tests

### ISSUE THREAD

**Issue by adrianbranescu** [Adrian-s-JavaScript-Programs/angular-locomotive-check-digit-generator#3]
In "locomotive.component.spec.ts" file:

1) At the beginning of some tests, the variable that holds the reference to a LocomotiveComponent instance is reassigned with the same value, although such an assignment is already done inside "beforeEach" method, which is executed before each unit test. Variable named "component" is declared in the "describe" block, assigned an explicit value in "beforeEach" method, then reassigned in some specs to point to the same reference.

2) In "beforeEach" asynchronous method, FormBuilder is added as provider, although it is not being used.

Tests and application work just fine, but unnecessary code should be removed.

**Comment by adrianbranescu** [comment]
Regarding the additional component fixture instance creation: 

Currently, in my code, some tests create that instance at the beginning, while other tests do not. All tests work, including when being run in random order.

I did some research, to make sure a component fixture instance already exists before each test, and I came to the conclusion that "beforeEach" method takes care of that.

From Jasmine official site: 
&nbsp;&nbsp;&nbsp;beforeEach(functionopt, timeoutopt)
&nbsp;&nbsp;&nbsp;Run some shared setup before each of the specs in the describe in which it is called.
https://jasmine.github.io/api/3.6/global

In my code, "beforeEach" method contains these lines:
```ts
    fixture = TestBed.createComponent(LocomotiveComponent);
    component = fixture.componentInstance;
```
That guarantees that each of my unit test will already have a component fixture instance available, so there is no need to create one again inside my specs.

But there are two "beforeEach" methods in my code: an asynchronous one and a synchronous one. Both of them were automatically generated by Angular CLI, and, after that, I updated them according to my needs.

I wanted to know what the relationship between these two "beforeEach" methods is, and that led me to Angular official site, where I found this paragraph:
<i>
A typical approach is to divide the setup logic into two separate beforeEach() functions:
    An async beforeEach() that compiles the components
    A synchronous beforeEach() that performs the remaining setup.
</i>

I also found this very important phrase:

<i>You can count on the test runner to wait for the first asynchronous beforeEach to finish before calling the second.</i>

If you wonder why the first "beforeEach" method is asynchronous, that is because it calls a method called "compileComponents", which requires to be called inside an asynchronous block.

Also, instead of two "beforeEach" methods, you can have one single one, asynchronous, where you chain the additional setup using "then" callback. It is called "Consolidated setup".

From Angular official site:

<i>You can consolidate the two beforeEach() functions into a single, async beforeEach().
The compileComponents() method returns a promise so you can perform the synchronous setup tasks after compilation by moving the synchronous code into a then(...) callback.</i>

Example:
```ts
beforeEach(waitForAsync(() => {
  TestBed
      .configureTestingModule({
        // initial setup here
      })
      .compileComponents()
      .then(() => {
        // remaining setup here
      });
}));
```
The above theory was found at this link: https://v10.angular.io/guide/testing-components-scenarios#the-async-beforeeach

"Consolidated setup" alternative is reinforcing the statement that you can count on the test runner to wait for the first asynchronous "beforeEach" method to finish before calling the synchronous "beforeEach" method.

After going through this theory, I can say it is safe to remove the ```component = fixture.componentInstance;``` lines at the beginning of the test specs where it appears. There is no need for this line, since setup inside "beforeEach" methods takes care of creating a new component instance before each test.

**Comment by adrianbranescu** [comment]
Regarding the "FormBuilder" issue:

In the early stages of unit tests writing, in "beforeEach" method I had something like this:
```ts
    component.locomotiveForm = formBuilder.group({
      serialNumber: null
    });
```
Since I was using "FormBuilder", I had to add it as provider when configuring TestBed.

Later, I switched to using "FormGroup", but "FormBuilder" remained as provider, although I was not using it anymore.

"FormBuilder" provider can now be removed from TestBed configuration, and also it should no longer be imported in "locomotive.component.spec.ts" file.

**Comment by adrianbranescu** [comment]
Regarding the redundant assignment:

component = fixture.componentInstance;

In the above code, "fixture" is of "ComponentFixture<LocomotiveComponent>" type and "component" is of "LocomotiveComponent" type. Also, "componentInstance" is a property (not a method!) of "ComponentFixture" class.

Currently, in the code, some tests make that assignment at the beginning, while other tests do not. All tests work, including when being run in random order.

I did some research, to make sure "component" variable already is a reference to a non-null "LocomotiveComponent" object before each test, and I came to the conclusion that "beforeEach" method takes care of that.

From Jasmine official site: 
&nbsp;&nbsp;&nbsp;beforeEach(functionopt, timeoutopt)
&nbsp;&nbsp;&nbsp;Run some shared setup before each of the specs in the describe in which it is called.
https://jasmine.github.io/api/3.6/global

In my code, "beforeEach" method contains (among other things) these lines:
```ts
    fixture = TestBed.createComponent(LocomotiveComponent);
    component = fixture.componentInstance;
```
That guarantees that each of my unit tests will already have a reference to an existing LocomotiveComponent instance, so there is no need to create one again inside my specs (specs already have access to "fixture" and "component" variables, because they are declared in the "describe" block that contains the specs).

But there are two "beforeEach" methods in my code: an asynchronous one and a synchronous one. Both of them were automatically generated by Angular CLI, and, after that, I updated them according to my needs.

I wanted to know what the relationship between these two "beforeEach" methods is (I was worried that only of them might be called before a test, depending on test synchronicity), and that led me to Angular official site, where I found this paragraph:
<i>
A typical approach is to divide the setup logic into two separate beforeEach() functions:
    An async beforeEach() that compiles the components
    A synchronous beforeEach() that performs the remaining setup.
</i>

I also found this very important phrase:

<i>You can count on the test runner to wait for the first asynchronous beforeEach to finish before calling the second.</i>

If you wonder why the first "beforeEach" method is asynchronous, that is because it calls a method called "compileComponents", which requires to be called inside an asynchronous block.

Also, instead of two "beforeEach" methods, you can have a single one, asynchronous, where you chain the additional setup using "then" callback. It is called "Consolidated setup".

From Angular official site:

<i>You can consolidate the two beforeEach() functions into a single, async beforeEach().
The compileComponents() method returns a promise so you can perform the synchronous setup tasks after compilation by moving the synchronous code into a then(...) callback.</i>

Example:
```ts
beforeEach(waitForAsync(() => {
  TestBed
      .configureTestingModule({
        // initial setup here
      })
      .compileComponents()
      .then(() => {
        // remaining setup here
      });
}));
```
The above theory was found at this link: https://v10.angular.io/guide/testing-components-scenarios#the-async-beforeeach

"Consolidated setup" alternative is reinforcing the statement that you can count on the test runner to wait for the first asynchronous "beforeEach" method to finish before calling the synchronous "beforeEach" method.

After going through this theory, I can say it is safe to remove the ```component = fixture.componentInstance;``` line at the beginning of the test specs where it appears. There is no need for this line, since setup inside "beforeEach" methods takes care of creating a new ComponentFixture and a LocomotiveComponent whose references can later be accessed inside any unit test.

**Comment by adrianbranescu** [comment]
Regarding the "FormBuilder" issue:

In the early stages of unit tests writing, in "beforeEach" method I had something like this:
```ts
    component.locomotiveForm = formBuilder.group({
      serialNumber: null
    });
```
Since I was using "FormBuilder", I had to add it as provider when configuring TestBed.

Later, I switched to using "FormGroup", but "FormBuilder" remained as provider, although I was not using it anymore.

"FormBuilder" provider can now be removed from TestBed configuration, and also it should no longer be imported in "locomotive.component.spec.ts" file.
